pykechain/models/scope.py

```diff
@@ -225,18 +226,14 @@ def _update_scope_project_team(self, action, role, user):
         role = check_enum(role, ScopeRoles, "role")
         user = check_text(user, "user")
 
-        users: List[Dict] = self._client._retrieve_users()["results"]
-        user_object: Dict = find(users, lambda u: u["username"] == user)
-        if user_object is None:
-            raise NotFoundError(f'User "{user}" does not exist')
-
+        retrieved_user: User = self._client.user(username=user["username"])
         url = self._client._build_url(f"scope_{action}_{role}", scope_id=self.id)
 
         response = self._client._request(
             "PUT",
             url,
             params=API_EXTRA_PARAMS[self.__class__.__name__.lower()],
-            data={"user_id": user_object["pk"]},
+            data={"user_id": retrieved_user.id},
```

Comment:
maybe we can do a follow up ticket on ke-chain backend that allow us to use the `retieved_user.uid`